src/ring_theory/ideal/operations.lean

```diff
@@ -1190,6 +1190,13 @@ le_antisymm (map_le_iff_le_comap.2 $ mul_le.2 $ λ r hri s hsj,
   by rw [← f.map_mul];
   exact mem_map_of_mem f (mul_mem_mul hri hsj))
 
+protected theorem map_pow (n : ℕ) : map f (I^n) = (map f I)^n :=
+begin
+  induction n with n hn,
+  { rw [pow_zero, one_eq_top, map_top, pow_zero, one_eq_top], },
+  { rw [pow_succ, map_mul, pow_succ, hn], },
+end
```

Comment:
Is `map f` a `monoid_hom`? That might be a nice way to prove this, and would give results about `finset.prod` too if those became useful in future.

Reply:
you were tagged in a question about this before in #9345 :b [here](https://github.com/leanprover-community/mathlib/pull/9345/files#r719195585)

Reply:
hmm, github seems to not be working, but just open up the dscussion near `map_pow` in that PR

Reply:
I added the submodule version of this result in #12374

Reply:
Since `submodule.map` and `ideal.map` are quite different, I decided to go ahead and make a bundled map hom in this PR without the dependency on #12374.